server/src/test/java/org/elasticsearch/index/codec/vectors/es816/ES816BinaryQuantizedVectorsWriter.java

```diff
@@ -513,12 +516,20 @@ private CloseableRandomVectorScorerSupplier mergeOneFieldToIndex(
                 ),
                 vectorValues
             );
+            final IndexOutput finalTempQuantizedVectorData = tempQuantizedVectorData;
+            final IndexOutput finalTempScoreQuantizedVectorData = tempScoreQuantizedVectorData;
+            success = true;
             return new BinarizedCloseableRandomVectorScorerSupplier(scorerSupplier, vectorValues, () -> {
-                IOUtils.close(finalBinarizedDataInput, finalBinarizedScoreDataInput);
+                IOUtils.close(
+                    finalBinarizedDataInput,
+                    finalBinarizedScoreDataInput,
+                    finalTempQuantizedVectorData,
+                    finalTempScoreQuantizedVectorData
```

Comment:
These are closed higher up in the function, but closing them twice is probably ok.

Reply:
ops, I didn't realize. Let's fix it so we don't close it twice.